parser/parser.go

```diff
@@ -59,60 +57,51 @@ type (
 
 func RunParser(reader io.Reader) Parser {
 	p := &pcmParser{
-		source:          reader,
-		keysReady:       make(chan struct{}),
-		keysInfoMutex:   new(sync.RWMutex),
-		streamInfoReady: make(chan struct{}),
-		streamInfoMutex: new(sync.RWMutex),
+		source: reader,
+		mutex:  new(sync.RWMutex),
 	}
 	go p.run()
-	return p
-}
 
-func RunStreamedParser(reader io.Reader, chanLen int) (Parser, <-chan ValuesOrError) {
-	p := &pcmParser{
-		source:        reader,
-		sink:          make(chan ValuesOrError, chanLen),
-		keysReady:     make(chan struct{}),
-		keysInfoMutex: new(sync.RWMutex),
-	}
-	go p.run()
+	sig := make(chan struct{})
+	go func() {
+		for {
```

Comment:
This loop has no resched or blocking operation (waiting for lock, channel, io etc.), so it can hang in unfavorable conditions.